### PR TITLE
envoy: Bump envoy version to 1.22.7

### PR DESCRIPTION
**Pull request by sayboras**
The runtime feature guard envoy.reloadable_features.no_extension_lookup_by_name
is set to false to avoid proxy crash for cilium.tls_wrapper.

https://www.envoyproxy.io/docs/envoy/latest/version_history/v1.22/v1.22.0.html

Relates: https://github.com/cilium/proxy/pull/101
Signed-off-by: Tam Mach <tam.mach@cilium.io>

### CODE REVIEW

**Comment by sayboras** [comment]
/test-1.16-4.9

**Comment by sayboras** [comment]
/test-1.26-net-next

**Comment by sayboras** [comment]
/test-1.25-4.19

**Comment by sayboras** [comment]
/test-1.26-net-next

**Comment by sayboras** [comment]
Rerun test-1.26-net-next and test-1.25-4.19 due to VM provision failure

**Comment by sayboras** [comment]
/test-1.26-net-next

**Comment by sayboras** [comment]
/test-1.25-4.19

**Comment by sayboras** [comment]
/test-1.24-5.4

**Comment by sayboras** [comment]
/test-1.16-4.9

**Review by rolinh — APPROVED** [review]

**Review by jrajahalme — APPROVED** [review]

**Comment by sayboras** [comment]
/test-1.24-5.4

**Comment by sayboras** [comment]
Reviews are in, all required CI jobs are passed except _Datapath BPF Complexity_ , which is not applicable. 

Mark this ready to merge.

File: images/cilium/Dockerfile [hunk 1]
```diff
@@ -6,7 +6,7 @@ ARG CILIUM_RUNTIME_IMAGE=quay.io/cilium/cilium-runtime:c1b4d66c0e1b08e95a1834df6
 
 # cilium-envoy from github.com/cilium/proxy
 #
-FROM quay.io/cilium/cilium-envoy:ff9a74a473dd5c1f6bfd7ce10938dab648f70c09@sha256:35afe14537ebf9fcfe4ab3d9488ec7f884bc2661da73383294f9b72deee82cfe as cilium-envoy
```
Comment (pchaigno) [inline images/cilium/Dockerfile:9]: @sayboras Is this actually 1.22.7? The commit says 1.22.6.

Comment (sayboras) [reply images/cilium/Dockerfile:9]: yes it is :), seems like I didn't update commit message after force push :(. The build is coming from https://github.com/cilium/proxy/actions/runs/4124721794/jobs/7124323004


Image publishing step

```
#18 pushing manifest for quay.io/cilium/cilium-envoy:d59b7f950565d4ccad5c9e30f725740271b7f1c2@sha256:a9c20ee37fe8b280606b096f34f074fed02a347381ded71f1853a214a324a9ef 0.2s done
#18 DONE 3.7s
```

Version check as part of build
```
./cilium-envoy version: d59b7f950565d4ccad5c9e30f725740271b7f1c2/1.22.7/Distribution/RELEASE/BoringSSL
```

**Pull request merged**